src/items/implementations.md

```diff
@@ -1,10 +1,62 @@
 # Implementations
 
-An _implementation_ is an item that can implement a [trait](items/traits.html) for a
-specific type.
+An _implementation_ is an item that associates items with an *implementing type*.
+
+There are two types of implementations: Bare implementations and [trait] 
+implementations.
 
 Implementations are defined with the keyword `impl`.
 
+## Bare Implementations
+
+A bare implementation is defined as the sequence of the `impl` keyword, generic
+type declarations, a path to a nomial tyupe, a where clause, and a bracketed
+set of associable items.
+
+The nominal type is called the *implementing type* and the associable items are
+the *associated items* to the implementing type.
```

Comment:
Hmm, libcore has `impl<T> [T] {`, where that slice isn't a nominal type.  Users cannot define structural types, though, so they can't do something like that...

Reply:
That definitely seems like a special case. Not entirely sure how to document that without messing up the flow for everything else.